service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/MedicationStatementMapper.java

```diff
@@ -296,7 +269,20 @@ private String buildParticipant(MedicationRequest medicationRequest) {
         var isOrganization = buildPredicateReferenceIsA(ResourceType.Organization);
         Predicate<Reference> isRelevant = isPractitioner.or(isPractitionerRole).or(isOrganization);
 
-        if (medicationRequest.hasRecorder() && medicationRequest.getRecorder().hasReference()) {
+        if (medicationRequest.hasRequester()) {
+            final var requester = medicationRequest.getRequester();
+            final var agent = requester.getAgent();
```

Comment:
You check for `if (medicationRequest.hasRequester()) {` but not for `requester.hasAgent()` ?